packages/dds/legacy-dds/src/array/sharedArray.ts

```diff
@@ -426,7 +426,24 @@ export class SharedArrayClass<T extends SerializableTypeForSharedArray>
 				}
 				break;
 			}
-			case OperationType.moveEntry:
+			case OperationType.moveEntry: {
+				const { entryId: oldEntryId, changedToEntryId: newEntryId } = arrayOp;
+				if (this.getEntryForId(newEntryId).isDeleted) {
+					return;
+				}
+				this.updateLiveEntry(newEntryId, oldEntryId);
+				const inputEntry = this.getEntryForId(oldEntryId);
+				inputEntry.prevEntryId = undefined;
+				inputEntry.nextEntryId = undefined;
+				inputEntry.isLocalPendingMove = 0;
+				const moveOp: IMoveOperation = {
+					type: OperationType.moveEntry,
+					entryId: oldEntryId,
+					changedToEntryId: newEntryId,
```

Comment:
is this correct? since this is rollback i would think the entryId should be newEntryIdm and changeTo should be oldEntryId since i believe we are reactivating the oldEntryId

Reply:
add some testing around events would help validate